### PR TITLE
best of n 연결 및 device model name arg추가

### PR DESCRIPTION
**Pull request by ksyint**

### CODE REVIEW

**Review by ksyint — COMMENTED** [review]
model_name 추가할필요가 없었을까요?....

**Comment by ksyint** [comment]
https://huggingface.co/heegyu/ko-reward-model-1.3b-v0.1
https://docs.vllm.ai/en/v0.6.5/models/pooling_models.html

한국어 전용 리워드모델인데 vllm 에 있는 리워드 모델들은 영어 위주이고 task = reward 할 경우 vector 값만 추출해서 위 heegyu모델로 대체하는 방안을 생각했습니다.

File: llm_eval/evaluator.py [hunk 1]
```diff
@@ -65,6 +65,11 @@ def run(
         model_params: Optional[Dict[str, Any]] = None,
         scaling_params: Optional[Dict[str, Any]] = None,
         evaluator_params: Optional[Dict[str, Any]] = None,
+        
+        
+        device:str=None,
```
Comment (h-albert-lee) [inline llm_eval/evaluator.py:70]: device는 huggingface backend에만 종속되어있는 args다보니 api단에서 다른 arg와 동 수준으로 있어야하는 항목은 아닐것같긴 합니다. 실제로 없어도  **self.model_backend_params 부분에서 넘어가긴 할거구요

File: llm_eval/models/huggingface.py [hunk 2]
```diff
@@ -23,11 +23,11 @@ class HuggingFaceModel(BaseModel):
         (ex: "logits": {"sum_log_prob": float, "token_log_probs": [...], "tokens": [...]} )
       - cot_parser를 통해 chain_of_thought와 최종 answer를 분리 (예: "Final Answer:" 구분 등).
     """
-
+#model_name_or_path: str="facebook/opt-350m",
     def __init__(
         self,
-        model_name_or_path: str,
```
Comment (h-albert-lee) [inline llm_eval/models/huggingface.py:29]: typehint를 제거한 이유가 있을까요?

File: llm_eval/scaling_methods/best_of_n.py [hunk 3]
```diff
@@ -10,10 +12,14 @@ class BestOfN(BaseScalingMethod):
     * "우수" 판단은 score_fn으로 할 수도 있고, default로 첫 번째만 택할 수도 있음.
     """
 
-    def __init__(self, model: BaseModel = None, n: int = 5, score_fn=None, **kwargs):
+    def __init__(self, model: BaseModel = None, n: int = 5, score_fn="heegyu/ko-reward-model-1.3b-v0.1", **kwargs):
```
Comment (h-albert-lee) [inline llm_eval/scaling_methods/best_of_n.py:15]: score_fn에 default 모델로 해당 모델이 있어야 하는 이유가 있나요?

File: run.py
Comment (h-albert-lee) [inline run.py:1]: 해당 파일은 왜 있는건가요?

**Review by h-albert-lee — CHANGES_REQUESTED** [review]

**Comment by h-albert-lee** [comment]
> https://huggingface.co/heegyu/ko-reward-model-1.3b-v0.1 https://docs.vllm.ai/en/v0.6.5/models/pooling_models.html
> 
> 한국어 전용 리워드모델인데 vllm 에 있는 리워드 모델들은 영어 위주이고 task = reward 할 경우 vector 값만 추출해서 위 heegyu모델로 대체하는 방안을 생각했습니다.

요거는 documentation에 내용을 추가해야겠네요

File: run.py
Comment (ksyint) [reply run.py:1]: 마찬가지로 삭제하겠습니다.

File: llm_eval/scaling_methods/best_of_n.py (hunk 3 above)
Comment (ksyint) [reply llm_eval/scaling_methods/best_of_n.py:15]: best of n 같은 경우 reward model 이 주는 스코어 가 기준이라 판단했는데, 리워드모델없이 best of n 할만한 metric을 추천해주시면 감사드리겠습니다. 아니면 리워드 모델을 args 단에서 선택하는것으로 구현해야 하는게 맞을지 질문드립니다.

File: llm_eval/models/huggingface.py (hunk 2 above)
Comment (ksyint) [reply llm_eval/models/huggingface.py:29]: 제가 모르고 지웠던거 같은데, 다시 확인 해보겠습니다.

File: llm_eval/evaluator.py (hunk 1 above)
Comment (ksyint) [reply llm_eval/evaluator.py:70]: 네 알겠습니다. 해당 내용 삭제하겠습니다.

**Comment by guijinSON** [comment]
> https://huggingface.co/heegyu/ko-reward-model-1.3b-v0.1 https://docs.vllm.ai/en/v0.6.5/models/pooling_models.html
> 
> 한국어 전용 리워드모델인데 vllm 에 있는 리워드 모델들은 영어 위주이고 task = reward 할 경우 vector 값만 추출해서 위 heegyu모델로 대체하는 방안을 생각했습니다.

영어 리워드 모델이 다른 언어에서 멀티링구얼하게 잘 작동한다는 theoretical, empirical 논문들이 많아서 굳이 필요 없을 것 같긴합니다.

영어 RM이 한국어 reward modeling을 잘한다는 emprical 증거 -> https://arxiv.org/abs/2409.11239
영어 base LM 에 한국어 reward modeling을 할 경우 일어나는 성능 저하에 대한 theoretical explanation -> https://arxiv.org/abs/2410.18027
영어 RM을 활용해 multilingual offline-rl 이 잘 작동한다는 empirical 주장 -> https://arxiv.org/abs/2404.12318
